### PR TITLE
Add FileNotFoundExceptions to the different YAML resolvers

### PR DESCRIPTION
**Pull request by tokee**
The current behaviour when trying to create a YAML from non-existing resources is to fail with `java.io.IOException: Stream closed` which could be a lot of things. This pull request makes the relevant methods throw proper `FileNotFoundException`s so that the caller knows what went wrong.

### CODE REVIEW

File: src/test/java/dk/kb/util/yaml/YAMLTest.java [hunk 1]
```diff
@@ -94,6 +96,33 @@ public void testMultiConfig() throws IOException {
...
     }
 
+    @Test
+    public void testFailingMultiConfig() throws IOException {
+        try {
+            YAML.resolveMultiConfig("Not_there.yml", "Not_there_2.yml");
+            fail("Resolving non-existing multi-config should not work");
+        } catch (FileNotFoundException e) {
+            // Expected
+        }
```
Comment (sbktc) [inline src/test/java/dk/kb/util/yaml/YAMLTest.java:106]: Perhaps utilising junit 5's expected exception may make it look cleaner?
See: https://howtodoinjava.com/junit5/expected-exception-example/

File: src/main/java/dk/kb/util/yaml/YAML.java [hunk 2]
```diff
@@ -651,20 +652,25 @@ public static YAML parse(InputStream yamlStream) {
...
     /**
      * Parse the given File as YAML.
      *
      * @param yamlFile path to YAML File.
      * @return a YAML based on the given stream.
+     * @throws IOException if a configuration could not be fetched.
+     * @throws FileNotFoundException if the given yamlFile could not be located.
      */
     public static YAML parse(File yamlFile) throws IOException {
+        if (!yamlFile.exists()) {
```
Comment (sbktc) [inline src/main/java/dk/kb/util/yaml/YAML.java:671]: While we are at it, should we consider the case where there file exists, but we are not allowed to read it?

**Review by sbktc — CHANGES_REQUESTED** [review]

File: src/test/java/dk/kb/util/yaml/YAMLTest.java (hunk 1 above)
Comment (tokee) [reply src/test/java/dk/kb/util/yaml/YAMLTest.java]: I did not know that one. Very nice!

I have updated `YAMLTest` to take advantage of the assert.

File: src/main/java/dk/kb/util/yaml/YAML.java (hunk 2 above)
Comment (tokee) [reply src/main/java/dk/kb/util/yaml/YAML.java:673]: Yep. Added together with a Unit-test.

**Review by sbktc — APPROVED** [review]

**Pull request merged**